tests/integration/integration_steps/cdf_steps.py

```diff
@@ -0,0 +1,119 @@
+import pandas as pd
+from datetime import datetime
+from time import sleep
+from cognite.client import CogniteClient
+from cognite.client.data_classes import Datapoint, TimeSeries, TimeSeriesWrite
+from cognite.client.exceptions import CogniteNotFoundError
+from cognite.client.data_classes import DataPointSubscriptionWrite
+
+def push_data_points_to_cdf(
+    external_id: str, data_points: list[Datapoint], cognite_client: CogniteClient
+) -> pd.DataFrame:
+    df = pd.DataFrame(columns=["timestamp", "value", "externalId"])
+
+    data_point_list = []
+
+    for datapoint in data_points:
+        data_point_list.append(
+            {
+                "externalId": external_id,
+                "timestamp": str(datapoint.timestamp),
+                "value": datapoint.value
+            }
+        )
+
+    df_original = pd.DataFrame(data_point_list)
+    df = df_original.pivot(index="timestamp", columns="externalId", values="value")
+    df.index = pd.to_datetime(df.index)
+
+    cognite_client.time_series.data.insert_dataframe(df)
+    return df_original
+
+
+def push_time_series_to_cdf(time_series_data: list[TimeSeries], cognite_client: CogniteClient) -> list[TimeSeries]:
+    time_series_write_list = []
+
+    for timeseries in time_series_data:
+        time_series_write_list.append(TimeSeriesWrite(
+            external_id=timeseries.external_id,
+            name=timeseries.name,
+            metadata=timeseries.metadata,
+            is_string=timeseries.is_string,
+            security_categories=timeseries.security_categories,
+            is_step=timeseries.is_step,
+            description=timeseries.description,
+        ))
+
+    cognite_client.time_series.create(time_series_write_list)
+    return time_series_data
+
+def push_data_to_cdf(time_series_data: list[TimeSeries], cognite_client: CogniteClient):
+    time_series_data_points_pushed = {}
+    for ts in time_series_data:
+        time_series_data_points_pushed[ts.external_id] = push_data_points_to_cdf(
+            data_points=ts.datapoints, external_id=ts.external_id, cognite_client=cognite_client
+        )
+    sleep(5)
+    return time_series_data_points_pushed
+
+def create_subscription_in_cdf(time_series_data: list[TimeSeries], sub_name: str, cognite_client: CogniteClient):
+    ts_external_ids = [ts.external_id for ts in time_series_data]
+    sub = DataPointSubscriptionWrite(sub_name, partition_count=1, time_series_ids=ts_external_ids, name="Test subscription")
+    return cognite_client.time_series.subscriptions.create(sub)
+
+def create_data_model_in_cdf():
+    # Create a data model in CDF
+    pass
+
+def update_data_model_in_cdf():
+    # Update a data model in CDF
+    pass
+
+
+def compare_timestamps(timestamp1: datetime, timestamp2: datetime) -> bool:
+    return timestamp1.replace(microsecond=0) == timestamp2.replace(microsecond=0)
```

Comment:
why are we ignoring microseconds in the comparison? Maybe add a comment why this is needed?

Reply:
The microsecond precision is lost to the thousandths of milliseconds at some point in the process so we're rounding to seconds.  I'll add a comment